src/cdk/overlay/overlay-ref.ts

```diff
@@ -83,11 +83,8 @@ export class OverlayRef implements PortalHost {
     return attachResult;
   }
 
-  /**
-   * Detaches an overlay from a portal.
-   * @returns Resolves when the overlay has been detached.
-   */
-  detach(): Promise<any> {
+  /** Detaches an overlay from a portal. */
+  detach(): void {
```

Comment:
Change the `PortalHost` signature as well?

Reply:
Not sure about this. I thought that the `PortalHost` can still have the type in case someone wants to implement his own `PortalHost`.

Reply:
But since you're removing the return value from the `OverlayRef`, it means that even if they did, they wouldn't be able to use it with our overlays anyway.

Reply:
Portals don't have to be used specifically with overlays.

Reply:
They don't but that's the primary use case and it's something that we'll end up not supporting. Since the initial goal of the PR was to get rid of the `Promise` signature, consider re-adding the return and setting the return value to `any`?

Reply:
That would work too. In that case we would have a very unclear method signature and the return value would be for now always `undefined`. 

Since overlays always seem to use the `DomPortalHost` which seems to return nothing on `detach` I thought we can be clear with this signature here.

I'm fine switching to `any` though.

Reply:
Updated as discussed on Slack.